src/Functions/fromReadableDecimalSize.cpp

```diff
@@ -0,0 +1,122 @@
+    .syntax = "fromReadableDecimalSize(x)",
+    .arguments = {{"x", "Readable size with decimal units ([String](../../sql-reference/data-types/string.md))"}},
+    .returned_value = "Number of bytes, rounded up to the nearest integer ([UInt64](../../sql-reference/data-types/int-uint.md))",
+    .examples = {
+        {
+            "basic",
+            "SELECT arrayJoin(['1 B', '1 KB', '3 MB', '5.314 KB']) AS readable_sizes, fromReadableDecimalSize(readable_sizes) AS sizes;",
+            R"(
+┌─readable_sizes─┬───sizes─┐
+│ 1 B            │       1 │
+│ 1 KB           │    1000 │
+│ 3 MB           │ 3000000 │
+│ 5.314 KB       │    5314 │
+└────────────────┴─────────┘)"
+        },
+    },
+    .categories = {"OtherFunctions"},
+};
+
+FunctionDocumentation fromReadableDecimalSizeOrNull_documentation {
+    .description = "Given a string containing the readable representation of a byte size with decimal units this function returns the corresponding number of bytes, or `NULL` if unable to parse the value.",
+    .syntax = "fromReadableDecimalSizeOrNull(x)",
+    .arguments = {{"x", "Readable size with decimal units ([String](../../sql-reference/data-types/string.md))"}},
+    .returned_value = "Number of bytes, rounded up to the nearest integer, or NULL if unable to parse the input (Nullable([UInt64](../../sql-reference/data-types/int-uint.md)))",
+    .examples = {
+        {
+            "basic",
+            "SELECT arrayJoin(['1 B', '1 KB', '3 MB', '5.314 KB', 'invalid']) AS readable_sizes, fromReadableSizeOrNull(readable_sizes) AS sizes;",
+            R"(
+┌─readable_sizes─┬───sizes─┐
+│ 1 B            │       1 │
+│ 1 KB           │    1000 │
+│ 3 MB           │ 3000000 │
+│ 5.314 KB       │    5314 │
+│ invalid        │    ᴺᵁᴸᴸ │
+└────────────────┴─────────┘)"
+        },
+    },
+    .categories = {"OtherFunctions"},
+};
+
+FunctionDocumentation fromReadableDecimalSizeOrZero_documentation {
+    .description = "Given a string containing the readable representation of a byte size with decimal units this function returns the corresponding number of bytes, or 0 if unable to parse the value.",
+    .syntax = "formatReadableSizeOrZero(x)",
+    .arguments = {{"x", "Readable size with decimal units ([String](../../sql-reference/data-types/string.md))"}},
+    .returned_value = "Number of bytes, rounded up to the nearest integer, or 0 if unable to parse the input ([UInt64](../../sql-reference/data-types/int-uint.md))",
+    .examples = {
+        {
+            "basic",
+            "SELECT arrayJoin(['1 B', '1 KB', '3 MB', '5.314 KB', 'invalid']) AS readable_sizes, fromReadableSizeOrZero(readable_sizes) AS sizes;",
+            R"(
+┌─readable_sizes─┬───sizes─┐
+│ 1 B            │       1 │
+│ 1 KB           │    1000 │
+│ 3 MB           │ 3000000 │
+│ 5.314 KB       │    5000 │
+│ invalid        │       0 │
+└────────────────┴─────────┘)"
+        },
+    },
+    .categories = {"OtherFunctions"},
+};
+}
```

Comment:
I've created `FunctionDocumentation` objects for each variant of the `fromReadable*Size`. The strings are quite but not completely similar over the 6 combinations ((Regular, Decimal)*(Exception, Null, Zero)) so I'm not sure if it's worth figuring out some sort of generalization like we have in `generateUUIDv7` or we can simply leave it as-is.

Reply:
Let's keep it simple and stupid and keep things as is.